File: intreverse.kt
/*class Solution {
    fun reverse(x: Int): Int {
        var y = x
        var r = 0
        while (y != 0) {
            val p = y % 10
            if ((r > 0 && r > (Int.MAX_VALUE - y) / 10) || (r < 0 && r < (Int.MIN_VALUE - y) / 10)) {
                return 0
            }
            r = r * 10 + p
            y /= 10
        }
        return r
    }
}*/
class Solution {
    fun reverse(x: Int): Int {
        var r = 0
        var y = x
        while (y != 0) {
             if (r <Int.MIN_VALUE / 10 || r > Int.MAX_VALUE / 10) {
            return 0
        }
        val digit = y % 10
        y /= 10
        r = r * 10 + digit
        }    
        return r
    }
}

File: intreverse.c
int reverse(int x)
{
       int num=0;
    while(x/10){
        num=num*10+x%10;
        x/=10;
    }
    if((num>INT_MAX/10)||((num==INT_MAX/10)&&(x>7)))   return 0;
    else if((num<INT_MIN/10)||((num==INT_MIN/10)&&(x<-8)))  return 0;
    else    return num*10+x;
}
/*int reverse(int x) {
    int rev = 0;
    while (x != 0) {
        if (rev < INT_MIN / 10 || rev > INT_MAX / 10) {
            return 0;
        }
        int digit = x % 10;
        x /= 10;
        rev = rev * 10 + digit;
    }
    return rev;
}*/
